XmlPullParserException: Binary XML file line #1: invalid drawable tag vector

#### Problem description

> FATAL EXCEPTION: main
                                                                         android.content.res.Resources$NotFoundException: File res/drawable/ic_place.xml from drawable resource ID #0x7f0800a8. If the resource you are trying to use is a vector resource, you may be referencing it in an unsupported way. See AppCompatDelegate.setCompatVectorFromResourcesEnabled() for more info.
                                                                             at android.content.res.Resources.loadDrawable(Resources.java:2842)
                                                                             at android.content.res.Resources.getDrawable(Resources.java:1521)
                                                                             at android.support.v4.content.ContextCompat.getDrawable(ContextCompat.java:360)
                                                                             at org.odk.collect.android.activities.GeoShapeOsmMapActivity$15.longPressHelper(GeoShapeOsmMapActivity.java:425)
                                                                             at org.osmdroid.views.overlay.MapEventsOverlay.onLongPress(MapEventsOverlay.java:51)
                                                                             at org.osmdroid.views.overlay.DefaultOverlayManager.onLongPress(DefaultOverlayManager.java:266)
                                                                             at org.osmdroid.views.MapView$MapViewGestureDetectorListener.onLongPress(MapView.java:1326)
                                                                             at android.view.GestureDetector.dispatchLongPress(GestureDetector.java:1508)
                                                                             at android.view.GestureDetector.access$200(GestureDetector.java:47)
                                                                             at android.view.GestureDetector$GestureHandler.handleMessage(GestureDetector.java:305)
                                                                             at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                             at android.os.Looper.loop(Looper.java:176)
                                                                             at android.app.ActivityThread.main(ActivityThread.java:5317)
                                                                             at java.lang.reflect.Method.invokeNative(Native Method)
                                                                             at java.lang.reflect.Method.invoke(Method.java:511)
                                                                             at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1102)
                                                                             at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:869)
                                                                             at dalvik.system.NativeStart.main(Native Method)
                                                                          Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #1: invalid drawable tag vector
                                                                             at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:917)
                                                                             at android.graphics.drawable.Drawable.createFromXml(Drawable.java:858)
                                                                             at android.content.res.Resources.loadDrawable(Resources.java:2839)

--- a/collect_app/src/main/java/org/odk/collect/android/activities/EditFormHierarchyActivity.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/EditFormHierarchyActivity.java
@@ -53,7 +53,7 @@ public class EditFormHierarchyActivity extends FormHierarchyActivity {
                 for (int i = 0; i < children.size(); i++) {
                     formList.remove(position + 1);
                 }
-                h.setIcon(ContextCompat.getDrawable(getApplicationContext(), R.drawable.expander_ic_minimized));
+                h.setIcon(ContextCompat.getDrawable(this, R.drawable.expander_ic_minimized));
                 break;
             case COLLAPSED:
                 Collect.getInstance().getActivityLogger().logInstanceAction(this, "onListItemClick",
@@ -65,7 +65,7 @@ public class EditFormHierarchyActivity extends FormHierarchyActivity {
                     formList.add(position + 1 + i, children1.get(i));
 
                 }
-                h.setIcon(ContextCompat.getDrawable(getApplicationContext(), R.drawable.expander_ic_maximized));
+                h.setIcon(ContextCompat.getDrawable(this, R.drawable.expander_ic_maximized));
                 break;
             case QUESTION:
                 Collect.getInstance().getActivityLogger().logInstanceAction(this, "onListItemClick",

--- a/collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java
@@ -339,7 +339,7 @@ public abstract class FormHierarchyActivity extends CollectAbstractActivity impl
                             // Display the repeat header for the group.
                             HierarchyElement group =
                                     new HierarchyElement(getLabel(fc), null, ContextCompat
-                                            .getDrawable(getApplicationContext(), R.drawable.expander_ic_minimized),
+                                            .getDrawable(this, R.drawable.expander_ic_minimized),
                                             COLLAPSED, fc.getIndex());
                             formList.add(group);
                         }

--- a/collect_app/src/main/java/org/odk/collect/android/activities/GeoPointOsmMapActivity.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/GeoPointOsmMapActivity.java
@@ -129,7 +129,7 @@ public class GeoPointOsmMapActivity extends CollectAbstractActivity implements L
         }
 
         marker = new Marker(map);
-        marker.setIcon(ContextCompat.getDrawable(getApplicationContext(), R.drawable.ic_place));
+        marker.setIcon(ContextCompat.getDrawable(this, R.drawable.ic_place));
         myLocationOverlay = new MyLocationNewOverlay(map);
 
         handler.postDelayed(new Runnable() {
@@ -485,7 +485,7 @@ public class GeoPointOsmMapActivity extends CollectAbstractActivity implements L
         showLocationButton.setEnabled(true);
         map.invalidate();
         marker.setPosition(geoPoint);
-        marker.setIcon(ContextCompat.getDrawable(getApplicationContext(), R.drawable.ic_place));
+        marker.setIcon(ContextCompat.getDrawable(this, R.drawable.ic_place));
         marker.setDraggable(true);
         latLng = geoPoint;
         isDragged = true;

--- a/collect_app/src/main/java/org/odk/collect/android/activities/GeoShapeOsmMapActivity.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/GeoShapeOsmMapActivity.java
@@ -236,7 +236,7 @@ public class GeoShapeOsmMapActivity extends CollectAbstractActivity implements I
             Marker marker = new Marker(map);
             marker.setPosition(new GeoPoint(gp[0], gp[1]));
             marker.setDraggable(true);
-            marker.setIcon(ContextCompat.getDrawable(getApplicationContext(), R.drawable.ic_place));
+            marker.setIcon(ContextCompat.getDrawable(this, R.drawable.ic_place));
             marker.setAnchor(Marker.ANCHOR_CENTER, Marker.ANCHOR_BOTTOM);
             marker.setOnMarkerClickListener(nullMarkerListener);
             mapMarkers.add(marker);
@@ -422,7 +422,7 @@ public class GeoShapeOsmMapActivity extends CollectAbstractActivity implements I
             Marker marker = new Marker(map);
             marker.setPosition(point);
             marker.setDraggable(true);
-            marker.setIcon(ContextCompat.getDrawable(getApplicationContext(), R.drawable.ic_place));
+            marker.setIcon(ContextCompat.getDrawable(GeoShapeOsmMapActivity.this, R.drawable.ic_place));
             marker.setAnchor(Marker.ANCHOR_CENTER, Marker.ANCHOR_BOTTOM);
             marker.setOnMarkerClickListener(nullMarkerListener);
             mapMarkers.add(marker);

--- a/collect_app/src/main/java/org/odk/collect/android/activities/GeoTraceOsmMapActivity.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/GeoTraceOsmMapActivity.java
@@ -393,7 +393,7 @@ public class GeoTraceOsmMapActivity extends CollectAbstractActivity implements I
             marker.setOnMarkerClickListener(nullMarkerListener);
             marker.setDraggable(true);
             marker.setOnMarkerDragListener(dragListener);
-            marker.setIcon(ContextCompat.getDrawable(getApplicationContext(), R.drawable.ic_place));
+            marker.setIcon(ContextCompat.getDrawable(this, R.drawable.ic_place));
             marker.setAnchor(Marker.ANCHOR_CENTER, Marker.ANCHOR_BOTTOM);
             mapMarkers.add(marker);
             List<GeoPoint> points = polyline.getPoints();
@@ -639,7 +639,7 @@ public class GeoTraceOsmMapActivity extends CollectAbstractActivity implements I
             Float lastKnownAccuracy =
                     myLocationOverlay.getMyLocationProvider().getLastKnownLocation().getAccuracy();
             myLocationOverlay.getMyLocationProvider().getLastKnownLocation().getAccuracy();
-            marker.setIcon(ContextCompat.getDrawable(getApplicationContext(), R.drawable.ic_place));
+            marker.setIcon(ContextCompat.getDrawable(this, R.drawable.ic_place));
             marker.setSubDescription(Float.toString(lastKnownAccuracy));
             marker.setAnchor(Marker.ANCHOR_CENTER, Marker.ANCHOR_BOTTOM);
             marker.setDraggable(true);

--- a/collect_app/src/main/java/org/odk/collect/android/activities/ViewFormHierarchyActivity.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/ViewFormHierarchyActivity.java
@@ -78,7 +78,7 @@ public class ViewFormHierarchyActivity extends FormHierarchyActivity {
                 for (int i = 0; i < children.size(); i++) {
                     formList.remove(position + 1);
                 }
-                h.setIcon(ContextCompat.getDrawable(getApplicationContext(), R.drawable.expander_ic_minimized));
+                h.setIcon(ContextCompat.getDrawable(this, R.drawable.expander_ic_minimized));
                 break;
             case COLLAPSED:
                 Collect.getInstance().getActivityLogger().logInstanceAction(this, "onListItemClick",
@@ -90,7 +90,7 @@ public class ViewFormHierarchyActivity extends FormHierarchyActivity {
                     formList.add(position + 1 + i, children1.get(i));
 
                 }
-                h.setIcon(ContextCompat.getDrawable(getApplicationContext(), R.drawable.expander_ic_maximized));
+                h.setIcon(ContextCompat.getDrawable(this, R.drawable.expander_ic_maximized));
                 break;
             case QUESTION:
                 Collect.getInstance().getActivityLogger().logInstanceAction(this, "onListItemClick",

